Gems/ROS2/Code/Source/Lidar/LidarTemplateUtils.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "LidarTemplateUtils.h"
+#include <AzCore/Math/MathUtils.h>
+#include <AzCore/Math/Matrix4x4.h>
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/std/containers/unordered_map.h>
+
+namespace ROS2
+{
+    LidarTemplate LidarTemplateUtils::GetTemplate(LidarTemplate::LidarModel model)
+    {
+        static std::unordered_map<LidarTemplate::LidarModel, LidarTemplate> templates;
+
+        if (templates.empty())
+        {
+            LidarTemplate generic3DLidar = { /*.m_model = */ LidarTemplate::Generic3DLidar,
+                                             /*.m_name = */ "GenericLidar",
+                                             /*.m_minHAngle = */ -180.0f,
+                                             /*.m_maxHAngle = */ 180.0f,
+                                             /*.m_minVAngle = */ 35.0f,
+                                             /*.m_maxVAngle = */ -35.0f,
+                                             /*.m_layers = */ 24,
+                                             /*.m_numberOfIncrements = */ 924,
+                                             /*.m_maxRange = */ 100.0f };
+            templates[LidarTemplate::Generic3DLidar] = generic3DLidar;
+        }
+
+        auto it = templates.find(model);
+        if (it == templates.end())
+        {
+            return LidarTemplate(); // TODO - handle it
+        }
+
+        return it->second;
+    }
+
+    size_t LidarTemplateUtils::TotalPointCount(const LidarTemplate& t)
+    {
+        return t.m_layers * t.m_numberOfIncrements;
+    }
+
+    // TODO - lidars in reality do not have uniform distributions - populating needs to be defined per model
```

Comment:
This [TODO](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#todo-comments) should be made into a GHI